twingate/client_resource.go

```diff
@@ -75,309 +113,233 @@ func validatePort(port string) (int64, error) {
-		resources[i] = c
+func (client *Client) readResource(resourceID graphql.ID) (*Resource, error) {
+	if resourceID.(string) == "" {
+		return nil, NewAPIErrorWithID(ErrGraphqlIDIsEmpty, "read", remoteNetworkResourceName, "resourceID")
 	}
 
-	return resources, nil
-}
-
-func (client *Client) readResource(resourceID string) (*Resource, error) { //nolint:funlen
-	mutation := map[string]string{
-		"query": fmt.Sprintf(`
-		{
-		  resource(id: "%s") {
-			id
-			name
-			address {
-			  type
-			  value
-			}
-			remoteNetwork {
-			  id
-			}
-			groups (first: 50){
-			  pageInfo{
-				hasNextPage
-			  }
-			  edges{
-				node {
-				  id
-				}
-			  }
-			}
-			protocols {
-			  udp {
-				ports {
-				  end
-				  start
-				}
-				policy
-			  }
-			  tcp {
-				ports {
-				  end
-				  start
-				}
-				policy
-			  }
-			  allowIcmp
-			}
-		  }
-		}
-        `, resourceID),
+	r := readResourceQuery{}
+	variables := map[string]interface{}{
+		"id":    resourceID,
+		"first": graphql.Int(readResourceQueryGroupsSize),
 	}
 
-	queryResource, err := client.doGraphqlRequest(mutation)
+	err := client.GraphqlClient.Query(context.Background(), &r, variables)
 	if err != nil {
-		return nil, NewAPIErrorWithID(err, "read", resourceResourceName, resourceID)
+		return nil, NewAPIErrorWithID(err, "read", remoteNetworkResourceName, resourceID)
 	}
 
-	resourceQuery := queryResource.Path("data.resource")
-	if resourceQuery.Data() == nil {
+	if r.Resource == nil {
 		return nil, NewAPIErrorWithID(err, "read", resourceResourceName, resourceID)
 	}
 
-	var groups = make([]string, 0)
+	var groups = make([]*graphql.ID, 0)
 
-	hasNextPage := resourceQuery.Path("groups.pageInfo.hasNextPage").Data().(bool)
-	if hasNextPage {
+	if r.Resource.Groups.PageInfo.HasNextPage {
 		return nil, NewAPIErrorWithID(ErrTooManyGroupsError, "read", resourceResourceName, resourceID)
 	}
 
-	for _, elem := range resourceQuery.Path("groups.edges").Children() {
-		nodeID := elem.Path("node.id").Data().(string)
-		groups = append(groups, nodeID)
+	for _, elem := range r.Resource.Groups.Edges {
+		groups = append(groups, &elem.Node.ID)
 	}
 
 	resource := &Resource{
-		ID:        resourceID,
-		Name:      resourceQuery.Path("name").Data().(string),
-		Address:   resourceQuery.Path("address.value").Data().(string),
-		GroupsIds: groups,
+		ID:              resourceID,
+		Name:            r.Resource.Name,
+		Address:         r.Resource.Address.Value,
+		RemoteNetworkID: r.Resource.RemoteNetwork.ID,
+		GroupsIds:       groups,
+		Protocols:       r.Resource.Protocols,
 	}
 
-	if resourceQuery.ExistsP("remoteNetwork.id") {
-		resource.RemoteNetworkID = resourceQuery.Path("remoteNetwork.id").Data().(string)
+	return resource, nil
+}
+
+type readResourcesQuery struct { //nolint
+	Resources struct {
+		Edges []*Edges
 	}
+}
 
-	extractProtocolsFromResult(resource, resourceQuery)
+func (client *Client) readResources() ([]*Edges, error) { //nolint
```

Comment:
why nolint?

Reply:
readResources, readRemoteNetworks, readConnectors are used only in sweep tests
So, it was marked as //nolint to prevent lint fails
```
 func `(*Client).readResources` is unused (unused)
```
i guess @shmykyta can explain it better because he implemented sweep tests.

Reply:
I current didnt find around it , it seems when a function is only called from tests the linter marks it , thus `nolint` 
this function is only called from sweeper tests